### DISCUSSION TITLE
Import & Use RuxButton

### DISCUSSION THREAD

**Discussion by Oceanestars** [RocketCommunicationsInc/astro#1267] [Q&A] [answered]
Hi!
I'm trying to follow the [docs](https://astro-components.netlify.app/?path=/docs/components-button--default-story) for the button component.

In react, I'm trying to import the button rux-button. And I get "Property 'rux-button' does not exist on type 'JSX.IntrinsicElements'." and it won't load the page/crash the page

```
import { RuxButton } from '@astrouxds/astro-web-components/dist/components/rux-button'
customElements.define('rux-button', RuxButton)

const container = () => {

return (
<div style="padding: 10%; display: flex; justify-content: center;">
    <rux-button>
        Button
    </rux-button>
</div>
)

```

Here is a [sandbox ](https://codesandbox.io/p/sandbox/magical-gwen-4z39qr?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cloz6iooh0008356h4tbdt24w%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cloz6iooh0003356h55no2bu4%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cloz6iooh0005356hxynviyjf%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cloz6iooh0007356hnctaxj9u%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cloz6iooh0003356h55no2bu4%2522%253A%257B%2522id%2522%253A%2522cloz6iooh0003356h55no2bu4%2522%252C%2522activeTabId%2522%253A%2522cloz6ivhi0078356huil4gdmj%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloz6iooh0002356hfspp9mzo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fvite.config.ts%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522cloz6ivhi0078356huil4gdmj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522cloz6iooh0007356hnctaxj9u%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloz6iooh0006356hcawghkxg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cloz6iooh0007356hnctaxj9u%2522%252C%2522activeTabId%2522%253A%2522cloz6iooh0006356hcawghkxg%2522%257D%252C%2522cloz6iooh0005356hxynviyjf%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloz6iooh0004356hh6lue47b%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522cloz6iooh0005356hxynviyjf%2522%252C%2522activeTabId%2522%253A%2522cloz6iooh0004356hh6lue47b%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)for easy reproduction.

If this is an issue I can convert this to an issue but I think it's more me missing something :)

Thanks for any help!

**Comment by micahjones13** [comment]
Hey @Oceanestars, is there any specific reason that you're importing the rux-button from the `@astrouxds/astro-web-components` package instead of the RuxButton from the `@astrouxds/react` package?  Doing the latter will solve your issue, as that imports the button component that's wrapped for React. 
Here's an updated [sandbox](https://codesandbox.io/p/sandbox/boring-antonelli-53z6xl?file=%2Fsrc%2FApp.tsx%3A12%2C13) for reference, but please let me know if I'm missing anything!

**Reply by Oceanestars** [reply]
Hey,
No,I copied and pasted the code from here https://astro-components.netlify.app/?path=/docs/components-button--default-story#cherry-picking

Will try that different import!

**Reply by micahjones13** [reply, marked as answer]
There are a few different packages available depending on what framework you're using, which can get confusing - and our docs don't do an amazing job at pointing that out on first glance, so I apologize. For future reference, `@astrouxds/react` is our component library wrapped for React, `@astrouxds/angular` is (you guessed it!) for Angular, and the `@astrouxds/astro-web-components` is the vanilla web components library which can be used with vanilla HTML/JS or Vue.

**Reply by Oceanestars** [reply]
That makes sense. Thanks! 
Also that fixed my issue. Wondering why adding onRuxChange on the RuxButton would give this warning(I believe I read somewhere OnRuxChange and onChange are interchangeable or is that incorrect?)

> Type '{ children: string; onRuxChange: void; }' is not assignable to type 'IntrinsicAttributes & RuxButton & Omit<HTMLAttributes<HTMLRuxButtonElement>, "style"> & StyleReactProps & RefAttributes<...>'.
>   Property 'onRuxChange' does not exist on type 'IntrinsicAttributes & RuxButton & Omit<HTMLAttributes<HTMLRuxButtonElement>, "style"> & StyleReactProps & RefAttributes<...>'. Did you mean 'onChange'?

**Reply by micahjones13** [reply]
The change events are a bit tricky because the vanilla `on change` and React's `onChange` fire at different times. The existing browser behavior fires the on change event when a user commits a change to an element, through something like a blur event. React on the other hand fires it's onChange event whenever a change is made to the element, like typing a new character into an input field. 
Because our components are web components, they follow the existing browser definition for the `on change` event. This means that the `onRuxchange` event is the browsers version of `on change`. The regular React `onChange` should work for most cases, but if you're running into issues with it I'd recommend trying `onRuxinput` to see if that solves any issues. 

Additionally, your problem may come from trying to add an `onChange` event to a button, which doesn't have that type of event. You may be looking for `onClick` in this case.
